pkg/mdformatter/transformer.go

```diff
@@ -40,7 +41,56 @@ func (t *transformer) Render(w io.Writer, source []byte, node ast.Node) error {
 	if err := ast.Walk(node, func(n ast.Node, entering bool) (ast.WalkStatus, error) {
 		var err error
 		switch typedNode := n.(type) {
-		// TODO(bwplotka): Add support for links inside HTML.
+		case *ast.HTMLBlock, *ast.RawHTML:
+			if !entering {
+				return ast.WalkSkipChildren, nil
+			}
+
+			// Parse HTML to get inline links on our own, goldmark does not do that.
+			b := bytes.Buffer{}
+			for i := 0; i < node.Lines().Len(); i++ {
+				segment := node.Lines().At(i)
+				_, _ = b.Write(segment.Value(source))
```

Comment:
Seems like node stays empty in this. So it is never tokenized and the first line of HTML block gets removed, which is also why `format not formatted` and `format formatted` tests both fail.